.github/workflows/release.yml

```diff
@@ -1,49 +1,110 @@
 name: release
 
+# Automate deployment to PyPi when creating a release tag vX.Y.Z
+# will only be published to PyPi if the git tag matches the release version
+# and the pre-commit and tests pass
+
 on:
-  release:
-    types: [published, edited]
+  push:
+    tags:
+      - "v*"
 
 jobs:
 
-  upload-transifex:
-    # Every time when a new version is released,
-    # upload the latest pot files to transifex services for team transilation.
-    # https://www.transifex.com/aiidateam/aiida-core/dashboard/
+  check-release-tag:
 
-    name: Upload pot files to transifex
     runs-on: ubuntu-latest
-    timeout-minutes: 30
 
-    # Build doc to pot files and register them to `.tx/config` file
-    # Installation steps are modeled after the docs job in `ci.yml`
     steps:
     - uses: actions/checkout@v2
-
-    - name: Set up Python 3.7
+    - name: Set up Python 3.8
       uses: actions/setup-python@v2
       with:
-        python-version: 3.7
+        python-version: 3.8
+    - run: python .github/workflows/check_release_tag.py $GITHUB_REF setup.json
```

Comment:
will this relative path of `setup.json` always work?

Reply:
yes the checkout action set the working directory as the repo root. We already assume this with things like `run: .github/workflows/tests.sh`